src/ILLink.RoslynAnalyzer/RequiresUnreferencedCodeAnalyzer.cs

```diff
@@ -50,6 +52,37 @@ public sealed class RequiresUnreferencedCodeAnalyzer : RequiresAnalyzerBase
 		private protected override ImmutableArray<(Action<OperationAnalysisContext> Action, OperationKind[] OperationKind)> ExtraOperationActions =>
 			ImmutableArray.Create ((s_dynamicTypeInvocation, new OperationKind[] { OperationKind.DynamicInvocation }));
 
+		protected override ImmutableArray<ISymbol> GetSpecialIncompatibleMembers (Compilation compilation)
+		{
+			var incompatibleMembers = ImmutableArray.CreateBuilder<ISymbol> ();
+			var typeType = compilation.GetTypeByMetadataName ("System.Type");
+			if (typeType != null) {
+				incompatibleMembers.AddRange (typeType.GetMembers ("MakeGenericType").OfType<IMethodSymbol> ());
+			}
+
+			var methodInfoType = compilation.GetTypeByMetadataName ("System.Reflection.MethodInfo");
+			if (methodInfoType != null) {
+				incompatibleMembers.AddRange (methodInfoType.GetMembers ("MakeGenericMethod").OfType<IMethodSymbol> ());
+			}
+
+			return incompatibleMembers.ToImmutable ();
+		}
+
+		protected override bool ReportSpecialIncompatibleMembersDiagnostic (OperationAnalysisContext operationContext, ImmutableArray<ISymbol> specialIncompatibleMembers, ISymbol member)
+		{
+			if (member is IMethodSymbol method && ImmutableArrayOperations.Contains (specialIncompatibleMembers, member, SymbolEqualityComparer.Default)) {
+				if (method.Name == "MakeGenericType") {
+					operationContext.ReportDiagnostic (Diagnostic.Create (s_makeGenericTypeRule, operationContext.Operation.Syntax.GetLocation (), method.GetDisplayName ()));
```

Comment:
I wonder if we should simply not warn here.
The guiding principal for the analyzer should be:
"If the analyzer produces a warning -> linker will also produce that warning"

In this case this will now hold - linker has more complex handling of MakeGenericType/Method which will avoid warning in many cases. The analyzer would warn every time. I would much rather we special case this and not warn for now, until we can build enough smarts in the analyzer to match the linker behavior.

/cc @sbomer - we discussed the general behavior, so for context.